react/BuyboxContext.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react'
+import { ProductContextProvider, useProduct } from 'vtex.product-context'
+import { defineMessages } from 'react-intl'
+
+const BuyboxContext: StorefrontFunctionComponent = ({ children }) => {
+  const productContext = useProduct() ?? {}
+
+  return (
+    <ProductContextProvider
+      query={{ skuId: productContext.selectedItem?.itemId }}
+      product={productContext.product}
+    >
+      {children}
+    </ProductContextProvider>
+  )
+}
+
+const messages = defineMessages({
+  title: {
+    id: 'admin/editor.buybox-context.title',
+  },
+})
+
+BuyboxContext.schema = {
+  title: messages.title.id,
+  type: 'object',
+}
+
+export default BuyboxContext
```

Comment:
Wouldn't it be possible to just import useProduct on every component? Because, if this app only exists in the product page, then it already has a ProductContextProvider, wouldn't this be redundant?

When you declare the main component as a block, you inset it into store.product block which already has the ProductContextProvider!

Does this make sense?

Reply:
For the next implementation I will need this structure, because I will replace the `sellers` that are inside `productContext` to set a new list of sellers sorted by something, you know?

Reply:
Oooh nice!!!